File: marmoraria-app/src/views/components/sidebar.py
import flet as ft
from src.config import COLOR_PRIMARY, COLOR_TEXT, COLOR_WHITE


class Sidebar(ft.Container):
    def __init__(self, page: ft.Page, is_mobile: bool = False):
        super().__init__()

        self.page = page

        # =========================
        # CONFIGURAÇÃO BASE
        # =========================
        self.bgcolor = COLOR_WHITE
        self.padding = 20

        if not is_mobile:
            self.width = 260
            self.border = ft.border.only(
                right=ft.BorderSide(1, "#F0F0F0")
            )

        # Papel do usuário
        user_role = getattr(self.page, "user_role", None)

        # =========================
        # LOGO / CABEÇALHO
        # =========================
        logo = ft.Container(
            content=ft.Row(
                [
                    ft.Container(
                        width=38,
                        height=38,
                        border_radius=10,
                        bgcolor=COLOR_PRIMARY,
                        alignment=ft.alignment.center,
                        content=ft.Icon(
                            ft.Icons.PRECISION_MANUFACTURING,
                            color=COLOR_WHITE,
                            size=20,
                        ),
                    ),
                    ft.Text(
                        "CENTRAL",
                        size=20,
                        weight=ft.FontWeight.BOLD,
                        color=COLOR_PRIMARY,
                    ),
                ],
                spacing=12,
            ),
            margin=ft.margin.only(bottom=30, left=5),
        )

        # =========================
        # ITENS DE MENU
        # =========================
        if user_role == "admin":
            menu_items = [
                self.criar_item_menu("Dashboard", ft.Icons.GRID_VIEW, "/dashboard"),
                self.criar_item_menu("Estoque", ft.Icons.INVENTORY_2, "/estoque"),
                self.criar_item_menu("Orçamentos", ft.Icons.RECEIPT_LONG, "/orcamentos"),
                self.criar_item_menu(
                    "Financeiro",
                    ft.Icons.ACCOUNT_BALANCE_WALLET,
                    "/financeiro",
                ),
                self.criar_item_menu(
                    "Produção",
                    ft.Icons.PRECISION_MANUFACTURING,
                    "/producao",
                ),
            ]
        else:
            menu_items = [
                self.criar_item_menu(
                    "Produção",
                    ft.Icons.PRECISION_MANUFACTURING,
                    "/producao",
                ),
            ]

        # =========================
        # RODAPÉ
        # =========================
        rodape = ft.Container(
            content=self.criar_item_menu(
                "Sair",
                ft.Icons.LOGOUT,
                "/login",
                estilo="danger",
            ),
            margin=ft.margin.only(top=10),
        )

        # =========================
        # CONTEÚDO FINAL
        # =========================
        self.content = ft.ListView(
            expand=True,
            spacing=10,
            controls=[
                logo,
                ft.Column(menu_items, spacing=5),
                ft.Divider(height=1, color="#F0F0F0"),
                rodape,
            ],
        )

    # ==================================================
    # NAVEGAÇÃO
    # ==================================================
    def navegar(self, e):
        rota = e.control.data

        # Fecha drawer no mobile
        if self.page.drawer:
            try:
                self.page.drawer.open = False
            except Exception:
                pass

        # Logout
        if rota == "/login":
            self.page.user_role = None

        self.page.go(rota)

    # ==================================================
    # ITEM DE MENU
    # ==================================================
    def criar_item_menu(self, texto, icone, rota, estilo="normal"):
        rota_atual = self.page.route
        esta_ativo = rota_atual == rota

        if estilo == "danger":
            cor_item = ft.Colors.RED_600
            bg_item = ft.Colors.RED_50 if esta_ativo else "transparent"
        else:
            cor_item = COLOR_PRIMARY if esta_ativo else ft.Colors.BLUE_GREY_700
            bg_item = f"{COLOR_PRIMARY}10" if esta_ativo else "transparent"

        return ft.Container(
            data=rota,
            on_click=self.navegar,
            padding=12,
            border_radius=12,
            bgcolor=bg_item,
            border=(
                ft.border.only(left=ft.BorderSide(3, COLOR_PRIMARY))
                if esta_ativo and estilo != "danger"
                else None
            ),
            content=ft.Row(
                [
                    ft.Icon(icone, size=22, color=cor_item),
                    ft.Text(
                        texto,
                        size=14,
                        weight=ft.FontWeight.BOLD if esta_ativo else ft.FontWeight.NORMAL,
                        color=cor_item,
                    ),
                ],
                spacing=15,
            ),
        )

File: lib/widgets/sidebar.dart
import 'package:flutter/material.dart';
import '../core/utils/constants.dart';

class Sidebar extends StatelessWidget {
  final String userRole;
  final String currentRoute;
  final bool isMobile;
  final VoidCallback onLogout;

  const Sidebar({
    super.key,
    required this.userRole,
    required this.currentRoute,
    this.isMobile = false,
    required this.onLogout,
  });

  @override
  Widget build(BuildContext context) {
    final menuItems = userRole == "admin"
        ? [
            _MenuItem("Dashboard", Icons.grid_view_rounded, "/dashboard"),
            _MenuItem("Estoque", Icons.inventory_2_outlined, "/estoque"),
            _MenuItem("Orçamentos", Icons.receipt_long_rounded, "/orcamentos"),
            _MenuItem("Financeiro", Icons.account_balance_wallet_rounded, "/financeiro"),
            _MenuItem("Produção", Icons.precision_manufacturing_outlined, "/producao"),
          ]
        : [
            _MenuItem("Produção", Icons.precision_manufacturing_outlined, "/producao"),
          ];

    return Container(
      width: isMobile ? null : 260,
      decoration: BoxDecoration(
        color: COLOR_WHITE,
        border: isMobile
            ? null
            : const Border(
                right: BorderSide(color: Color(0xFFF0F0F0), width: 1),
              ),
        boxShadow: isMobile ? null : const [SHADOW_MD],
      ),
      padding: EdgeInsets.fromLTRB(15, 25, 15, isMobile ? 20 : 10),
      child: Column(
        children: [
          _buildHeader(),
          const SizedBox(height: 30),
          Expanded(
            child: Column(
              children: menuItems
                  .map((item) => _buildMenuItem(context, item))
                  .toList(),
            ),
          ),
          const Divider(height: 1),
          const SizedBox(height: 10),
          _buildLogout(context),
        ],
      ),
    );
  }

  Widget _buildHeader() {
    return Row(
      children: [
        Container(
          width: 38,
          height: 38,
          decoration: BoxDecoration(
            color: COLOR_PRIMARY,
            borderRadius: BorderRadius.circular(10),
          ),
          child: const Icon(
            Icons.precision_manufacturing_rounded,
            color: Colors.white,
            size: 20,
          ),
        ),
        const SizedBox(width: 12),
        const Text(
          "CENTRAL",
          style: TextStyle(
            fontSize: 20,
            fontWeight: FontWeight.bold,
            color: COLOR_PRIMARY,
            letterSpacing: 1,
          ),
        ),
      ],
    );
  }

  Widget _buildMenuItem(BuildContext context, _MenuItem item, {bool danger = false}) {
    final isActive = currentRoute == item.route;

    final Color itemColor = danger
        ? Colors.red.shade600
        : (isActive ? COLOR_PRIMARY : Colors.blueGrey.shade700);

    // CORREÇÃO: Lógica do hoverBg consertada usando operador ternário
    final Color hoverBg = danger
        ? Colors.red.shade50
        : COLOR_PRIMARY.withValues(alpha: 0.1);

    return InkWell(
      onTap: () {
        if (!isActive) {
          Navigator.pushReplacementNamed(context, item.route);
        }
      },
      borderRadius: BorderRadius.circular(12),
      child: Container(
        margin: const EdgeInsets.symmetric(vertical: 4),
        padding: const EdgeInsets.symmetric(vertical: 12, horizontal: 15),
        decoration: BoxDecoration(
          color: isActive ? hoverBg : Colors.transparent,
          borderRadius: BorderRadius.circular(12),
          border: isActive && !danger
              ? const Border(
                  left: BorderSide(color: COLOR_PRIMARY, width: 3),
                )
              : null,
        ),
        child: Row(
          children: [
            Icon(item.icon, color: itemColor, size: 22),
            const SizedBox(width: 15),
            Text(
              item.label,
              style: TextStyle(
                fontSize: 14,
                fontWeight: isActive ? FontWeight.bold : FontWeight.w500,
                color: itemColor,
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildLogout(BuildContext context) {
    return InkWell(
      onTap: () {
        onLogout();
        Navigator.pushNamedAndRemoveUntil(context, "/login", (route) => false);
      },
      borderRadius: BorderRadius.circular(12),
      child: Container(
        padding: const EdgeInsets.symmetric(vertical: 12, horizontal: 15),
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(12),
        ),
        child: Row(
          children: [
            Icon(Icons.logout_rounded, color: Colors.red.shade600, size: 22),
            const SizedBox(width: 15),
            Text(
              "Sair do Sistema",
              style: TextStyle(
                fontSize: 14,
                fontWeight: FontWeight.w500,
                color: Colors.red.shade600,
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class _MenuItem {
  final String label;
  final IconData icon;
  final String route;

  _MenuItem(this.label, this.icon, this.route);
}